### PR TITLE
Unlock AWS provider v4. Require minimum 3.75.1.

### PR DESCRIPTION
**Pull request by rhoboat**
## Description

Bumps aws provider version and adds an upgrade test.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

**Unlock AWS provider v4. Require minimum 3.75.1.** This update includes a few tests that make sure upgrading to this module from the last release is easy. However, you may need to bump your AWS provider version. See the migration guide notes below for more.

### Migration Guide

The AWS Provider v4 unlock is a functionally backward compatible update. Modules no longer have the AWS Provider v4 lock. Upgrade tests were run to give reasonable confidence that upgrading to this version of the modules from the last tagged release is backward compatible, requiring no further modifications from you. However, the AWS Provider version must be `3.75.1` or newer, so you may need to run `terraform init` with the `-upgrade` flag, which will allow terraform to pull the latest version of the AWS provider, as in `terraform init -upgrade`.

### CODE REVIEW

File: test/upgrades/upgrade_module_join_path_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	"github.com/gruntwork-io/module-ci/test/upgrades"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+)
+
+func TestUpgradeModuleJoinPath(t *testing.T) {
+	upgrades.RunUpgradeModuleTest(
+		t,
+		repoName,
+		"join-path",
```
Comment (yorinasub17) [inline test/upgrades/upgrade_module_join_path_test.go:25]: NIT: We probably want upgrade testing for the `instance-type` module moreso than `join-path`.

**Review by yorinasub17 — APPROVED** [review]
LGTM! Just one comment about which module would benefit the most from upgrade testing.

File: test/upgrades/upgrade_module_join_path_test.go (hunk 1 above)
Comment (rhoboat) [reply test/upgrades/upgrade_module_join_path_test.go:25]: Ah, thank you.

Comment (rhoboat) [reply test/upgrades/upgrade_module_join_path_test.go:25]: Added a test 3b65d6a..54b8d64.

**Review by yorinasub17 — APPROVED** [review]
LGTM!

**Comment by rhoboat** [comment]
Super duper thanks for the review!

**Pull request merged**